### PR TITLE
Randomize RandomNumberGenerator instances' default seed

### LINKED ISSUE #66989: RandomNumberGenerator is constructed with a fixed seed, but documentation states it is randomized
### Godot version

v4.0.beta.custom_build [0c23a2cfe]

### System information

Fedora 36

### Issue description

When constructing an RNG with `RandomNumberGenerator.new()`, the seed is always set to `-6398989897141750821`.
However, according to its classref description:

> The default values of seed and state properties are pseudo-random

Either the documentation needs to be fixed, or this is a bug that needs to be fixed, I'm not sure which. The GlobalScope random number generator is randomized implicitly now in 4.0, so I thought maybe the intention was for this to be too?

### Steps to reproduce

```gdscript
var random := RandomNumberGenerator.new()
print(random.seed)
print(random.state)
```
Prints `-6398989897141750821` and `-6332740101472756719` every time, even when constructing several generators.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Anutrix**
Fixes https://github.com/godotengine/godot/issues/66989.

### CODE REVIEW

File: doc/classes/RandomNumberGenerator.xml [hunk 1]
```diff
@@ -10,7 +10,6 @@
 		[codeblock]
 		var rng = RandomNumberGenerator.new()
 		func _ready():
-		    rng.randomize()
 		    var my_random_number = rng.randf_range(-10.0, 10.0)
 		[/codeblock]
 		[b]Note:[/b] The default values of [member seed] and [member state] properties are pseudo-random, and changes when calling [method randomize]. The [code]0[/code] value documented here is a placeholder, and not the actual default seed.
```
Comment (akien-mga) [inline doc/classes/RandomNumberGenerator.xml:15]:
```suggestion
		[b]Note:[/b] The default values of [member seed] and [member state] properties are pseudo-random, and change when calling [method randomize]. The [code]0[/code] value documented here is a placeholder, and not the actual default seed.
```

Comment (Anutrix) [reply doc/classes/RandomNumberGenerator.xml]: Done.

**Comment by Anutrix** [comment]
Can we merge this for 4.0 since it's better to break compat in 4.0 than 4.1?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**